### PR TITLE
adding eraser ai detector

### PR DESCRIPTION
**Pull request by 0x1**
AI Copilot for creating technical design: https://www.eraser.io/ai

### Description:
Add EraserAI detector

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/eraser/eraser.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+			isVerified, extraData, verificationErr := verifyMatch(ctx, client, match)
+			s1.Verified = isVerified
+			s1.ExtraData = extraData
+			s1.SetVerificationError(verificationErr, match)
+		}
+
+		results = append(results, s1)
+	}
+
+	return
```
Comment (abmussani) [inline pkg/detectors/eraser/eraser.go:66]: I guess you have missed to return the results with error ( probably nil).

**Review by abmussani — CHANGES_REQUESTED** [review]
Thank you @0x1 for adding this. With little change, You have also not added new detector in [pkg/engine/defaults.go](https://github.com/trufflesecurity/trufflehog/blob/main/pkg/engine/defaults.go).

File: pkg/detectors/eraser/eraser.go (hunk 1 above)
Comment (rosecodym) [reply pkg/detectors/eraser/eraser.go:66]: This is actually a [Go language feature](https://go.dev/tour/basics/7)! It's kind of confusing sometimes, but it has its uses, so we allow it.

**Review by ahrav — APPROVED** [review]

File: pkg/detectors/eraser/eraser.go (hunk 1 above)
Comment (abmussani) [reply pkg/detectors/eraser/eraser.go:66]: ahan, I did not know about 'naked returns'. Thank you @rosecodym for increasing my Golang knowledge.

**Pull request merged**